### PR TITLE
[docker] Refactor "autoscaler" image into "-autoscaler" tag and "ray-ml" image.

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?

This is further work at cleaning up/reorganizing our docker images. Part of this PR is moving the true "autoscaler" dependencies into `base-deps` and `ray-deps` (they can be optionally removed from those images, but will (for the time being) always be included. The other major change is renaming the `autoscaler` image to `ray-ml` as it has many ML libraries that `Tune` and `RLLib` require/operate with.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: ci/travis/build-docker-images.sh [hunk 1]
```diff
@@ -71,23 +71,25 @@ if [[ "$TRAVIS" == "true" ]]; then
 
     commit_sha=$(echo "$TRAVIS_COMMIT" | head -c 6)
     cp -r "$ROOT_DIR"/.whl "$ROOT_DIR"/docker/ray/.whl
-    cp "$ROOT_DIR"/python/requirements.txt "$ROOT_DIR"/docker/autoscaler/requirements.txt
-    cp "$ROOT_DIR"/python/requirements_autoscaler.txt "$ROOT_DIR"/docker/autoscaler/requirements_autoscaler.txt
+    cp "$ROOT_DIR"/python/requirements.txt "$ROOT_DIR"/docker/ray-ml/requirements.txt
+    cp "$ROOT_DIR"/python/requirements_ml_docker.txt "$ROOT_DIR"/docker/ray-ml/requirements_ml_docker.txt
+    cp "$ROOT_DIR"/python/requirements_tune.txt "$ROOT_DIR"/docker/ray-ml/requirements_tune.txt
+    cp "$ROOT_DIR"/python/requirements_rllib.txt "$ROOT_DIR"/docker/ray-ml/requirements_rllib.txt
```
Comment (simon-mo) [inline ci/travis/build-docker-images.sh:77]: I think glob syntax should still work? `cp "$ROOT_DIR"/python/requirements*.txt "$ROOT_DIR"/docker/ray-ml/`

File: docker/ray-deps/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,18 @@
 ARG GPU=""
+ARG AUTOSCALER="autoscaler"
```
Comment (simon-mo) [inline docker/ray-deps/Dockerfile:2]: Let's add a comment saying if this is removed, then autoscaler dependencies won't be installed

**Review by simon-mo — APPROVED** [review]
LGTM. Let's tag and upload the image to both `rayproject/autoscaler` and `rayproject/ray-ml` for few months to keep backward compatible.

File: docker/base-deps/Dockerfile [hunk 3]
```diff
@@ -31,10 +33,20 @@ RUN apt-get update -y && apt-get upgrade -y \
         cython==0.29.0 \
         numpy==1.15.4 \
         psutil \
+        blist \ 
+    # blist is needed for numpy (which is re-installed when ray is installed)
```
Comment (ijrsvt) [inline docker/base-deps/Dockerfile:37]: Remove from requirements.txt

File: ci/travis/build-docker-images.sh [hunk 4]
```diff
@@ -71,23 +71,25 @@ if [[ "$TRAVIS" == "true" ]]; then
...
 
     build_or_pull_base_images
 
 
     build_and_push_tags "ray" "$commit_sha"
 
-    build_and_push_tags "autoscaler" "$commit_sha"
+    build_and_push_tags "ray-ml" "$commit_sha"
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.sh:85]: Tag as rayproject/autoscaler as well!

File: ci/travis/build-docker-images.sh (hunk 1 above)
Comment (ijrsvt) [reply ci/travis/build-docker-images.sh:77]: Yep! That should do it!

File: docker/ray-deps/Dockerfile (hunk 2 above)
Comment (ijrsvt) [reply docker/ray-deps/Dockerfile:2]: Okay!

**Pull request merged**